core/src/main/java/me/mykindos/betterpvp/core/logging/menu/CachedLogMenu.java

```diff
@@ -0,0 +1,205 @@
+            LogContext.CLAN,
+            LogContext.CLIENT
+    );
+
+    public static final List<String> ITEM = List.of(
+            "All",
+            LogContext.CLIENT_NAME,
+            LogContext.ITEM,
+            LogContext.ITEM_NAME,
+            LogContext.CLIENT
+    );
+
+
+    private final String title;
+    private final String key;
+    private final String value;
+    private final @Nullable String actionFilter;
+    private final BPvPPlugin plugin;
+    private final LogRepository logRepository;
+    private IStringFilterButton actionButton;
+    private IStringFilterButton categoryButton;
+    private IStringFilterValueButton valueButton;
+
+    public CachedLogMenu(@NotNull String title, String key, String value, @Nullable String actionFilter, List<String> contexts, BPvPPlugin plugin, LogRepository logRepository, Windowed previous) {
+        super(9, 5, false, new Structure(
+                "# # # # # # A C V",
+                "# x x x x x x x #",
+                "# x x x x x x x #",
+                "# x x x x x x x #",
+                "# # # < - > # # R")
+                .addIngredient('x', Markers.CONTENT_LIST_SLOT_HORIZONTAL)
+                .addIngredient('#', Menu.BACKGROUND_ITEM)
+                .addIngredient('<', new PreviousButton())
+                .addIngredient('-', new BackButton(previous))
+                .addIngredient('>', new ForwardButton())
+                .addIngredient('R', new RefreshButton<>())
+                .addIngredient('A', new StringFilterButton<>("Select Action", 9))
+                .addIngredient('C', new StringFilterButton<>("Select Category", contexts, 9))
+                .addIngredient('V', new StringFilterValueButton<>(9)
+                )
+        );
+
+        if (getItem(8, 4) instanceof IRefreshButton refreshButton) {
+            refreshButton.setRefresh(this::refresh);
+        }
+
+        if (getItem(6, 0) instanceof IStringFilterButton filterButton) {
+            this.actionButton = filterButton;
+            filterButton.setRefresh(this::refresh);
+        }
+
+        if (getItem(7, 0) instanceof IStringFilterButton filterButton) {
+            this.categoryButton = filterButton;
+            filterButton.setRefresh(this::refresh);
+        }
+
+        if (getItem(8, 0) instanceof IStringFilterValueButton logContextFilterValueButton) {
+            this.valueButton = logContextFilterValueButton;
+            logContextFilterValueButton.setRefresh(this::refresh);
+        }
+
+
+        this.title = title;
+        this.key = key;
+        this.value = value;
+        this.actionFilter = actionFilter;
+        this.plugin = plugin;
+        this.logRepository = logRepository;
+        setContent(List.of(new SimpleItem(ItemView.builder()
+                .material(Material.PAPER)
+                .displayName(Component.text("Loading..."))
+                .build())
+        ));
+        refresh();
+    }
+
+    private CompletableFuture<Boolean> refresh() {
```

Comment:
@Mykindos @braulio-dev is this an acceptable way to handle refreshing the logs? Is this an ok way to filter, or do I need to think of a better approach?

Reply:
done